src/client/connection.rs

```diff
@@ -51,6 +51,12 @@ pub(crate) struct Connection {
     buf: BytesMut,
 }
 
+impl Debug for Connection {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "Connection")
```

Comment:
A better option here could be to use [`fmt::DebugStruct`](https://doc.rust-lang.org/std/fmt/struct.DebugStruct.html):

```suggestion
        f.debug_struct("Connection").finish()
```